CMakeLists.txt

```diff
@@ -46,6 +46,8 @@ endif()
 include(CXX1x)
 check_for_cxx17_compiler(CXX17_COMPILER)
 
+set(CMAKE_CXX_VISIBILITY_PRESET hidden)
```

Comment:
I would leave this for 1.14; also, if we make it default for `cmake`, then it should also become default for `make`
just create a `cmake` and a `make` CI job with hidden visibility and we'll remove those jobs once we make hidden default

Reply:
Added make and cmake jobs for macOS with visibility set to hidden.